src/main/scala/net/kaliber/play/prerender/PrerenderActionBuilders.scala

```diff
@@ -40,13 +40,24 @@ case class PrerenderActionBuilders(config: PrerenderConfig)(implicit ec: Executi
   }
 
   private def shouldBePrerendered(request: RequestHeader) =
-    config.enabled && isGetRequest(request) && isSearchEngineRequest(request)
+    config.enabled && isGetRequest(request) && isSearchEngineRequest(request) &&
+    hasOnlyAllowedParams(request) && !isExcludedWithRegex(request)
 
   private def isGetRequest(request: RequestHeader) = request.method == "GET"
 
   private def isSearchEngineRequest(request: RequestHeader) =
     isEscapedFragmentUrl(request) || hasSearchEngineUserAgent(request)
 
+  private def hasOnlyAllowedParams(request: RequestHeader) = {
+    val requestParams = request.queryString.keys.toSet
+    requestParams subsetOf config.limitedParams.getOrElse(requestParams)
+  }
+
+  private def isExcludedWithRegex(request: RequestHeader) =
+    config.excludeRegex
```

Comment:
I've made a change here according to EECOLOR's suggestion. Notice I took the empty string check out, and will now be expecting an empty string value for config.excludeRegex to exclude everything.